CRM/Financial/BAO/FinancialItem.php

```diff
@@ -286,25 +286,23 @@ public static function checkContactPresent($contactIds, &$error) {
   }
 
   /**
-   * Get last financial item data.
+   * Get non-liability financial item with the highest id relating to the line item.
    *
-   * @param int $entityId
+   * In practical terms this means return the row that is not associated with
+   * sales tax as sales tax is a liability account.
    *
-   * @param string $entityTable
+   * @param int $entityId
    *
    * @return object CRM_Core_DAO
    */
-  public static function getPreviousFinancialItem($entityId, $entityTable = 'civicrm_line_item') {
```

Comment:
I'd prefer to revert this since we want to support extensions using other entity tables.